### PR TITLE
test: add TS types check

### PR DESCRIPTION
**Pull request by sheerlox**
## Description
Imported and adapted the [types test file from `@types/cron`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/34872661a2fd0a6a43fe8da464bc2ec07ad49169/types/cron/cron-tests.ts).

*Note: The tests won't pass until #689 is merged since they detect the issue it fixes.*

## Related Issue
- #688

## Motivation and Context
This adds more safety to our workflow regarding TypeScript types. Note that this only statically checks the types, so there is no assurance the runtime types would match.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] If my change introduces a breaking change, I have added a `!` after the type/scope in the title (see the Conventional Commits standard).

### CODE REVIEW

File: types/index.test-d.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+// using factory function with parameters object
+expectType<cron.CronJob>(cron.job({
+  cronTime: '00 30 11 * * 1-5',
+  onTick: () => {},
+  start: false,
+  timeZone: 'America/Los_Angeles'
+}));
+
+// with timezone
+new CronJob(
```
Comment (intcreator) [inline types/index.test-d.ts:29]: with these instantiations that aren't assigned to variables, is the idea that the test will fail if there's a type mismatch?

Comment (sheerlox) [reply types/index.test-d.ts:29]: Yes absolutely, that file is never actually ran, it's only statically analysed by the Typescript compiler

**Review by intcreator — APPROVED** [review]

**Pull request merged**